REST Repositories for Models

RyanWork opened the pull request:
I'm not sure if we should be committing the target folder, but I noticed Kshamina used a template and I just kind of assumed it was right??? Not sure on this one, no clear cut answer from what I could find on Google.

Also, I put the CustomizedRestMvcConfiguration.java file in the root package. I wasn't really sure where to put it. Any suggestions? Or is leaving it at root okay?

RyanWork committed:
Added repositories for all of the model classes

RyanWork committed:
Added a class to change the basepath of rest calls

RyanWork committed:
Other shenanigans changed by intellij & building

SupriyaGadigone commented on src/main/java/SYSC4806/CustomizedRestMvcConfiguration.java:
```diff
@@ -0,0 +1,23 @@
+        config.setBasePath("/api");
+    }
+}
+
```
I think for now the location of this is okay here. I checked out some other repos and forums and people have been putting them either in the root or just the src package. So this should be fine.

adamstaples reviewed (approved):
What a legend, looks great!

andrewdlqnguyen reviewed (approved):
I cant seem to find anything online, besides some maven related implementations but from the look of the solutions, their suggestions are to put in src package. 
src/resource?  Think what you did is fine!

For the dependencies, could we put <version>RELEASE</version> under those that has without version? Just to keep the dependencies updated?

RyanWork committed:
Merged master to feature branch

RyanWork committed:
Changed the api routing to use application properties

RyanWork commented:
@SupriyaGadigone I just removed the original rerouting to the API and added a shorter thing to application.properties instead. :^) Probably can just accept it, but Kshamina told me to review EVVVVVVVVVVVERYTHING
